src/lib/expression/between_expression.cpp

```diff
@@ -6,30 +6,91 @@
 
 namespace opossum {
 
-BetweenExpression::BetweenExpression(const std::shared_ptr<AbstractExpression>& value,
-                                     const std::shared_ptr<AbstractExpression>& lower_bound,
-                                     const std::shared_ptr<AbstractExpression>& upper_bound)
-    : AbstractPredicateExpression(PredicateCondition::Between, {value, lower_bound, upper_bound}) {}
+PredicateCondition BaseBetweenExpression::get_between_predicate_expression(bool left_inclusive, bool right_inclusive) {
```

Comment:
Thus function is only used in the rule, I think it should be there instead of here. We have to try to keep the Expression interfaces small and clean